pkg/ebpf/tracer.go

```diff
@@ -549,9 +551,11 @@ func (t *Tracer) getConnections(active []network.ConnectionStats) ([]network.Con
 	key, stats := &ConnTuple{}, &ConnStatsWithTimestamp{}
 	seen := make(map[ConnTuple]struct{})
 	var expired []*ConnTuple
+	connExists, cleanup := connExistsFn(t.config.ProcRoot)
+	defer cleanup()
 	entries := mp.IterateFrom(unsafe.Pointer(&ConnTuple{}))
 	for entries.Next(unsafe.Pointer(key), unsafe.Pointer(stats)) {
-		if stats.isExpired(latestTime, t.timeoutForConn(key)) {
+		if stats.isExpired(latestTime, t.timeoutForConn(key)) && !connExists(key) {
```

Comment:
this is going to incur a netlink call per expired connection which might become a problem:

* have we done any performance testing on this?
* did you consider adding some kind of caching here to reduce that overhead or the overhead of constructing a new instance of the conntrack handle every time we call connections()
* should we check the userspace conntrack table before consulting the netlink copy

Reply:
I agree that at the very least it might be worth considering caching one netlink socket per namespace.

Reply:
I did consider having a more long-lived cache for the netlink connections. We'd have to put in some limits though, and an eviction policy when we reach that limit. TTL based cache?

Reply:
I think a LRU cache would be well-suited for this. hopefully we have one available already

Reply:
I think you mentioned TTL because we'll likely have a lot of short-lived namespaces? Makes sense too. I think both would be OK.

Reply:
So I wrote a cache where the TTL gets renewed on access. Let me put it up for review.